### PR TITLE
Bug 11537: avoid reset tenant in startup service

### PR DESCRIPTION
**Pull request by laedanrex**
## Description

BUG - Problème d'accès aux APPs Collecte et Contrat de gestion dans le cas d'une organisation multi-tenant

## Type de changement:

* Correction

* Refactorisation de code

## Tests:

manuel

## Contributeur

VAS (Vitam Accessible en Service)

### CODE REVIEW

**Comment by TDevillechabrolle** [comment]
Logo
**Checkmarx One – Scan Summary & Details** – [538296f9\-1ee0\-4cb7\-a3ab\-42ab9948f95f](https://eu.ast.checkmarx.net/projects/78091106-f137-4b7b-8b68-77ee0f4d3433/scans?id=538296f9-1ee0-4cb7-a3ab-42ab9948f95f&branch=vas-11537-bug-collect-user-multi-tenant)
### New Issues
Severity|Issue|Source File / Package|Checkmarx Insight
---|---|---|---
MEDIUM|[CVE-2023-2976](https://devhub.checkmarx.com/cve-details/CVE-2023-2976)|Maven\-com\.google\.guava:guava\-29\.0\-jre|[Vulnerable Package](https://eu.ast.checkmarx.net/sca/#/projects/78091106-f137-4b7b-8b68-77ee0f4d3433/reports/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/vulnerabilities/CVE-2023-2976%3AMaven-com.google.guava%3Aguava-29.0-jre/vulnerabilityDetails)

### Fixed Issues
Severity|Issue|Source File / Package|Checkmarx Insight
---|---|---|---
MEDIUM|[Privacy_Violation](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/sast/description/359/12956636075206043460)|/api/api\-iam/iam\-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService\.java: [150](https://github.com/ProgrammeVitam/vitam-ui/blob/vas-11537-bug-collect-user-multi-tenant//api/api-iam/iam-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService.java#L150)|[Attack Vector](https://eu.ast.checkmarx.net/results/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=kscBSNfsLifnc1Kq3YpGrN10pks%3D&redirect=true)
MEDIUM|[Privacy_Violation](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/sast/description/359/12956636075206043460)|/api/api\-iam/iam\-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService\.java: [110](https://github.com/ProgrammeVitam/vitam-ui/blob/vas-11537-bug-collect-user-multi-tenant//api/api-iam/iam-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService.java#L110)|[Attack Vector](https://eu.ast.checkmarx.net/results/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=aUZHycZS4msMZ7wz80hQsi2aeIk%3D&redirect=true)
MEDIUM|[Privacy_Violation](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/sast/description/359/12956636075206043460)|/api/api\-iam/iam\-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService\.java: [118](https://github.com/ProgrammeVitam/vitam-ui/blob/vas-11537-bug-collect-user-multi-tenant//api/api-iam/iam-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService.java#L118)|[Attack Vector](https://eu.ast.checkmarx.net/results/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=SfY8pRwd4Uv%2BFoLS%2B0693atnVuo%3D&redirect=true)
MEDIUM|[Privacy_Violation](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/sast/description/359/12956636075206043460)|/api/api\-iam/iam\-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService\.java: [150](https://github.com/ProgrammeVitam/vitam-ui/blob/vas-11537-bug-collect-user-multi-tenant//api/api-iam/iam-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService.java#L150)|[Attack Vector](https://eu.ast.checkmarx.net/results/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=VJiqSAzXMN7C%2BTdtpsMMR88YyW4%3D&redirect=true)
MEDIUM|[Privacy_Violation](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/sast/description/359/12956636075206043460)|/api/api\-iam/iam\-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService\.java: [110](https://github.com/ProgrammeVitam/vitam-ui/blob/vas-11537-bug-collect-user-multi-tenant//api/api-iam/iam-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService.java#L110)|[Attack Vector](https://eu.ast.checkmarx.net/results/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=a8B9U8DD9QCNL6oT9kSUozbAMsE%3D&redirect=true)
MEDIUM|[Privacy_Violation](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/sast/description/359/12956636075206043460)|/api/api\-iam/iam\-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService\.java: [118](https://github.com/ProgrammeVitam/vitam-ui/blob/vas-11537-bug-collect-user-multi-tenant//api/api-iam/iam-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService.java#L118)|[Attack Vector](https://eu.ast.checkmarx.net/results/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=SX2jXxGs4a5lUTCmovc6FyioulY%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/sast/description/117/9884244256835554923)|/ui/ui\-archive\-search/src/main/java/fr/gouv/vitamui/archives/search/rest/ArchivesSearchController\.java: [262](https://github.com/ProgrammeVitam/vitam-ui/blob/vas-11537-bug-collect-user-multi-tenant//ui/ui-archive-search/src/main/java/fr/gouv/vitamui/archives/search/rest/ArchivesSearchController.java#L262)|[Attack Vector](https://eu.ast.checkmarx.net/results/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=IYIKOLWXU8gEMts3kOT%2B92X9kX8%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/sast/description/117/9884244256835554923)|/ui/ui\-archive\-search/src/main/java/fr/gouv/vitamui/archives/search/rest/ArchivesSearchController\.java: [248](https://github.com/ProgrammeVitam/vitam-ui/blob/vas-11537-bug-collect-user-multi-tenant//ui/ui-archive-search/src/main/java/fr/gouv/vitamui/archives/search/rest/ArchivesSearchController.java#L248)|[Attack Vector](https://eu.ast.checkmarx.net/results/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=Y0b6YyqS7FSKErfGvMp3xhEqIhQ%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/sast/description/117/9884244256835554923)|/ui/ui\-archive\-search/src/main/java/fr/gouv/vitamui/archives/search/rest/ArchivesSearchController\.java: [286](https://github.com/ProgrammeVitam/vitam-ui/blob/vas-11537-bug-collect-user-multi-tenant//ui/ui-archive-search/src/main/java/fr/gouv/vitamui/archives/search/rest/ArchivesSearchController.java#L286)|[Attack Vector](https://eu.ast.checkmarx.net/results/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=ua6ZWBfEKTXjfAdySk43H%2FpS%2FOU%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/sast/description/117/9884244256835554923)|/ui/ui\-archive\-search/src/main/java/fr/gouv/vitamui/archives/search/rest/ArchivesSearchController\.java: [210](https://github.com/ProgrammeVitam/vitam-ui/blob/vas-11537-bug-collect-user-multi-tenant//ui/ui-archive-search/src/main/java/fr/gouv/vitamui/archives/search/rest/ArchivesSearchController.java#L210)|[Attack Vector](https://eu.ast.checkmarx.net/results/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=vIw%2BqTgY7FotMBwLWg0tvfoXtd0%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/sast/description/117/9884244256835554923)|/ui/ui\-archive\-search/src/main/java/fr/gouv/vitamui/archives/search/rest/ArchivesSearchController\.java: [262](https://github.com/ProgrammeVitam/vitam-ui/blob/vas-11537-bug-collect-user-multi-tenant//ui/ui-archive-search/src/main/java/fr/gouv/vitamui/archives/search/rest/ArchivesSearchController.java#L262)|[Attack Vector](https://eu.ast.checkmarx.net/results/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=thVOFkgP%2BwbslEMjj2SQk%2BASZqE%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/sast/description/117/9884244256835554923)|/ui/ui\-archive\-search/src/main/java/fr/gouv/vitamui/archives/search/rest/ArchivesSearchController\.java: [248](https://github.com/ProgrammeVitam/vitam-ui/blob/vas-11537-bug-collect-user-multi-tenant//ui/ui-archive-search/src/main/java/fr/gouv/vitamui/archives/search/rest/ArchivesSearchController.java#L248)|[Attack Vector](https://eu.ast.checkmarx.net/results/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=x2XRZSfOGU%2FwF1aaE825eIrk0UQ%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/sast/description/117/9884244256835554923)|/ui/ui\-archive\-search/src/main/java/fr/gouv/vitamui/archives/search/rest/ArchivesSearchController\.java: [298](https://github.com/ProgrammeVitam/vitam-ui/blob/vas-11537-bug-collect-user-multi-tenant//ui/ui-archive-search/src/main/java/fr/gouv/vitamui/archives/search/rest/ArchivesSearchController.java#L298)|[Attack Vector](https://eu.ast.checkmarx.net/results/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=KseAlmJkz%2Fc4Na5bhzG5cac6W%2Fs%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/sast/description/117/9884244256835554923)|/ui/ui\-archive\-search/src/main/java/fr/gouv/vitamui/archives/search/rest/ArchivesSearchController\.java: [286](https://github.com/ProgrammeVitam/vitam-ui/blob/vas-11537-bug-collect-user-multi-tenant//ui/ui-archive-search/src/main/java/fr/gouv/vitamui/archives/search/rest/ArchivesSearchController.java#L286)|[Attack Vector](https://eu.ast.checkmarx.net/results/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=yRtq3Qkhs0duaXcGNm3gInbcyRo%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/sast/description/117/9884244256835554923)|/api/api\-collect/collect\-internal/src/main/java/fr/gouv/vitamui/collect/internal/server/rest/TransactionArchiveUnitInternalController\.java: [162](https://github.com/ProgrammeVitam/vitam-ui/blob/vas-11537-bug-collect-user-multi-tenant//api/api-collect/collect-internal/src/main/java/fr/gouv/vitamui/collect/internal/server/rest/TransactionArchiveUnitInternalController.java#L162)|[Attack Vector](https://eu.ast.checkmarx.net/results/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=h3eM8c63kajP0qWmtFofe7cFlxU%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/sast/description/117/9884244256835554923)|/ui/ui\-archive\-search/src/main/java/fr/gouv/vitamui/archives/search/rest/ArchivesSearchController\.java: [111](https://github.com/ProgrammeVitam/vitam-ui/blob/vas-11537-bug-collect-user-multi-tenant//ui/ui-archive-search/src/main/java/fr/gouv/vitamui/archives/search/rest/ArchivesSearchController.java#L111)|[Attack Vector](https://eu.ast.checkmarx.net/results/538296f9-1ee0-4cb7-a3ab-42ab9948f95f/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=E6J8GNAH2d33INfmmmW0nwjVM38%3D&redirect=true)

File: ui/ui-frontend-common/src/app/modules/startup.service.ts [hunk 1]
```diff
@@ -120,16 +121,26 @@ export class StartupService {
...
 
+  /**
+   * @deprecated: use tenant-selection.service
+   */
   setTenantIdentifier(tenantIdentifier?: string) {
-    this.CURRENT_TENANT_IDENTIFIER = tenantIdentifier;
+    if (tenantIdentifier) {
+      this.CURRENT_TENANT_IDENTIFIER = tenantIdentifier;
+    } else {
+      // tenant not reset
```
Comment (mohatizaoui) [inline ui/ui-frontend-common/src/app/modules/startup.service.ts]: supprime le else

si deprectaed pourquoi tu as changé le code il y a des contraintes ?

File: ui/ui-frontend-common/src/app/modules/vitamui-http-interceptor.ts [hunk 2]
```diff
@@ -92,7 +93,8 @@ export class VitamUIHttpInterceptor implements HttpInterceptor {
     this.initSnackBarService();
 
     let tenantIdentifier = request.headers.get('X-Tenant-Id');
-    if ((!tenantIdentifier || tenantIdentifier === undefined || tenantIdentifier === '') && this.authService.user) {
+    if ((!tenantIdentifier || tenantIdentifier === '') && this.authService.user) {
+      // TODO: change to tenant-selection.service
```
Comment (Regzox) [inline ui/ui-frontend-common/src/app/modules/vitamui-http-interceptor.ts:97]: Service existant ou travail à planifier ?

**Review by Regzox — APPROVED** [review]

File: ui/ui-frontend-common/src/app/modules/startup.service.ts (hunk 1 above)
Comment (laedanrex) [reply ui/ui-frontend-common/src/app/modules/startup.service.ts]: c'est deprecated pour éviter qu'il ne soit utilisé à l'avenir mais il est toujours utilisé pour l'instant donc je ne peux pas le retirer.           
pour les tenant je prefererais que l'on utilise tenant-selection.service.   
le else je l'ai ajouté volontairement pour faire aparaitre le pb je peux le retirer

File: ui/ui-frontend-common/src/app/modules/vitamui-http-interceptor.ts (hunk 2 above)
Comment (laedanrex) [reply ui/ui-frontend-common/src/app/modules/vitamui-http-interceptor.ts:97]: service existant

**Review by mohatizaoui — APPROVED** [review]

**Review by mohatizaoui — APPROVED** [review]

**Pull request merged**